bvm/ballerina-profiler/src/main/java/io/ballerina/runtime/profiler/Main.java

```diff
@@ -0,0 +1,367 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com) All Rights Reserved.
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.runtime.profiler;
+
+import io.ballerina.runtime.profiler.codegen.ProfilerClassLoader;
+import io.ballerina.runtime.profiler.codegen.ProfilerMethodWrapper;
+import io.ballerina.runtime.profiler.util.Constants;
+import io.ballerina.runtime.profiler.util.ProfilerException;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.FilenameUtils;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.PrintWriter;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.jar.JarFile;
+import java.util.stream.Collectors;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+import static io.ballerina.runtime.profiler.ui.HttpServer.initializeHTMLExport;
+import static io.ballerina.runtime.profiler.ui.JsonParser.initializeCPUParser;
+import static io.ballerina.runtime.profiler.util.Constants.OUT_STREAM;
+
+/**
+ * This class is used to as the driver class of the Ballerina profiler.
+ *
+ * @since 2201.8.0
+ */
+public class Main {
+
+    static long profilerStartTime;
+    private static String balJarArgs = null;
+    static String balJarName = null;
+    static String targetDir = null;
+    static String skipFunctionString = null;
+    private static int balFunctionCount = 0;
+    static int moduleCount = 0;
+
+    static final List<String> INSTRUMENTED_PATHS = new ArrayList<>();
```

Comment:
Do we need capitalize variable names here?

Reply:
May be we can rename Main.Java to Profiler.Java and remove all static methods and variables.. Have a separate class called Main.java. Then we can use new instance of profiler and call some method like run.